nixos/modules/services/monitoring/prometheus/alertmanager.nix

```diff
@@ -0,0 +1,117 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alertmanager;
+  mkConfigFile = pkgs.writeText "alertmanager.yml" cfg.configuration;
+in {
+  options = {
+    services.alertmanager = {
+      enable = mkEnableOption "Prometheus Alertmanager";
+
+      user = mkOption {
+        type = types.str;
+        default = "root";
+        description = ''
+          User name under which Alertmanager shall be run.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "root";
+        description = ''
+          Group under which Alertmanager shall be run.
+        '';
+      };
+
+      configuration = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Alertmanager configuration.
+        '';
+      };
+
+      logFormat = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          If set use a syslog logger or JSON logging.
+        '';
+      };
+
+      logLevel = mkOption {
+        type = types.enum ["debug" "info" "warn" "error" "fatal"];
+        default = "warn";
+        description = ''
+    	    Only log messages with the given severity or above. Valid levels: [debug, info, warn, error, fatal].
+        '';
+      };
+
+      webExternalUrl = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+    	    The URL under which Alertmanager is externally reachable (for example, if Alertmanager is served via a reverse proxy).
+          Used for generating relative and absolute links back to Alertmanager itself.
+          If the URL has a path portion, it will be used to prefix all HTTP endoints served by Alertmanager.
+          If omitted, relevant URL components will be derived automatically.
+        '';
+      };
+
+      webListenAddress = mkOption {
```

Comment:
I'd prefer `listenAddress` and `port` to keep it consistent with other services.